### PR TITLE
MINOR: Fix stream-join metadata

### PR DESCRIPTION
**Pull request by vvcephei**
https://github.com/apache/kafka/pull/11356 inadvertently changed
the (undefined) header forwarding behavior of stream-stream joins.

This change does not define the behavior, but just restores the prior
undefined behavior for continuity's sake. Defining the header-forwarding
behavior is future work.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by vvcephei** [comment]
ping @jeqo , in case you have a chance to review as well.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java [hunk 1]
```diff
@@ -887,6 +892,98 @@ public void shouldNotEmitLeftJoinResultForAsymmetricAfterWindow() {
...
+            ));
+
+            // Again, header forwarding is undefined, but the current observed behavior is that
+            // the headers pass through the forwarding record.
+            processor.checkAndClearProcessedRecords(
+                new Record<>(
+                    1,
+                    "null+a0",
+                    0L,
+                    new RecordHeaders(new Header[]{new RecordHeader("h", new byte[]{0x3})})
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java:952]: Semantically really bad to forward `0x3`, but well, it is what it is.

File: streams/src/test/java/org/apache/kafka/test/MockApiProcessor.java [hunk 2]
```diff
@@ -92,6 +93,20 @@ public void process(final Record<KIn, VIn> record) {
     }
 
     public void checkAndClearProcessResult(final KeyValueTimestamp<?, ?>... expected) {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/test/MockApiProcessor.java:95]: Seems we should get rid of this one after we have updated the full DSL to use the new `Record` type.

**Review by mjsax — APPROVED** [review]

**Review by jeqo — APPROVED** [review]
LGTM, thanks,  @vvcephei!

**Review by jeqo — APPROVED** [review]

**Comment by vvcephei** [comment]
It seems like the build is getting better. Only a couple of unrelated test failures:
```
[Build / JDK 8 and Scala 2.12 / kafka.server.ListOffsetsRequestTest.testCurrentEpochValidation()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11952/1/testReport/junit/kafka.server/ListOffsetsRequestTest/Build___JDK_8_and_Scala_2_12___testCurrentEpochValidation__/)
    Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.integration.StreamsUncaughtExceptionHandlerIntegrationTest.shouldShutdownMultipleThreadApplication
```

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java (hunk 1 above)
Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java:952]: Yeah, I was a bit bummed to see this, but at least now we know!

File: streams/src/test/java/org/apache/kafka/test/MockApiProcessor.java (hunk 2 above)
Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/test/MockApiProcessor.java:95]: We certainly could, but there's also nothing semantically wrong with making KVT-based assertions, so I figured I'd just leave them both in the test infrastructure.

**Comment by vvcephei** [comment]
Thanks for the quick reviews, @mjsax and @jeqo !

**Pull request merged**